### DISCUSSION TITLE
Code Snippets not working anymore

### DISCUSSION THREAD

**Discussion by frankli0324** [community/community#54598] [Projects and Issues]
### Select Topic Area

Bug

### Select Feature Area

- [X] Issues
- [ ] Projects

### Body

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet

when pasting permalink in issues, they doesn't seem to convert into code blocks anymore?

for example: https://github.com/microsoft/vscode/issues/181676

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by frankli0324** [comment]
is there any issue to track this?

**Comment by posoo** [comment]
Same here. It doesn't work in either the issue body or the following comments, no matter the code snippet is public or private.

**Reply by yezhengli-Mr9** [reply]
Notice permalinks/ code snippets still work for commits' code diff comments, for example:
![image](https://github.com/community/community/assets/16505983/df5812b8-1f73-4abf-a44d-3ec61440bee1)
I am with personal free account private repo. Not work for issues.

**Comment by cipri-tom** [comment]
Yes, it's a shame it doesn't work anymore, it was a nice feature !

**Comment by yezhengli-Mr9** [comment]
For personal repo, code snippets/ permalinks work when referring to codes within same repo, not when referring to codes within other repos.

**Reply by heatblayze** [reply]
This doesn't seem to work for me. I can only see code snippets from comments on pull requests. Pasting a permalink to a file in the same repo onto an issue or PR just shows a hyperlink.

**Reply by kuzdogan** [reply]
Same here. idk why this does not work anymore

**Reply by MagnusOxlund** [reply]
I can't get this to work in a forked repo (yes, I'm sure I'm referencing code in the forked repo, not code in the original repo).

**Comment by valentijnscholten** [comment]
I ran into as well and noticed that it seems to be because of using URLs pointing to the HEAD of a branch instead of a permalink. When using VSCode and selecting 'Open file on remote', it brings you to GitHub in the browser and displays the file and highlights the line.
All looks OK. But that link is to the main/master branch for example and if pasted somewhere, it will not expand.

For example https://github.com/DependencyTrack/dependency-track/blob/master/src/test/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzerTest.java#L164

When on GitHub viewing the line/file, a permalink needs to be created to allow gitHub to expand it. This makes sense as the HEAD of a branch may change over time. This will work: https://github.com/DependencyTrack/dependency-track/blob/bcd26514f100f53554a3e35b003ecf9a6c0bdc26/src/test/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzerTest.java#L164

**Reply by ghost** [reply]
> I ran into as well and noticed that it seems to be because of using URLs pointing to the HEAD of a branch instead of a permalink. When using VSCode and selecting 'Open file on remote', it brings you to GitHub in the browser and displays the file and highlights the line. All looks OK. But that link is to the main/master branch for example and if pasted somewhere, it will not expand.
> 
> For example https://github.com/DependencyTrack/dependency-track/blob/master/src/test/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzerTest.java#L164
> 
> When on GitHub viewing the line/file, a permalink needs to be created to allow gitHub to expand it. This makes sense as the HEAD of a branch may change over time. This will work: https://github.com/DependencyTrack/dependency-track/blob/bcd26514f100f53554a3e35b003ecf9a6c0bdc26/src/test/java/org/dependencytrack/tasks/repositories/NugetMetaAnalyzerTest.java#L164

Mark this as outdated. Thanks!

**Reply by SuperKenVery** [reply]
Doesn't seem to work:

<img width="943" height="327" alt="图片" src="https://github.com/user-attachments/assets/eba2b337-ee59-42e0-be1b-f8707cfc562c" />

**Comment by ghost** [comment]
Status update for Apr -

Will work for single line permalinks. not for Multiline ones. Pretty obvious how to fix it. But currently you have to paste the link into a script to fix it. 

Edit - This is outdated now, why? Test for yourself when a multiline one stops working. Pick any public repo.

Edit - It's not reproducible in the exact same conditions. This means something changed and there is an unknown condition involved. Which explains why this hasn't been fixed, it can't be fixed by traditional methods anymore(you can also hypothesize).

---

Manish
<details><summary>Details</summary>
<p>

Check out my kind of oldfangled, kinda newfangled, kinda cool, inevitably thought-provoking profile. And the limited mentorship opportunities for your no matter the non business major you have chosen be it - Bachelor's or Master's or Doctorates or Post-Doctorates.
In short, if you want to do check out my Readme.md at https://github.com/ket395 or https://github.com/ket395/ket395. It's quite good, I declare. 

</p>
</details>

**Comment by cferreira-fuze** [comment]
It seems that for repos where the link was added months ago, it works. But for new links, it does not work.

Does the link requires to be permanent? Or can it reference the master branch also? I'm asking because despite the age, that's the second difference between the two cases mentioned above.
